chapters/classes.tex

```diff
@@ -383,14 +383,27 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 definition equations is given in \cref{variability-of-expressions}.
 
 \begin{nonnormative}
-A discrete-time variable is a piecewise constant signal which
-changes its values only at event instants during simulation. Such types
-of variables are needed in order that special algorithms, such as the
-algorithm of Pantelides for index reduction, can be applied (it must be
-known that the time derivative of these variables is identical to zero).
-Furthermore, memory requirements can be reduced in the simulation
-environment, if it is known that a component can only change at event
-instants.
+For \lstinline!Real! variables we can distinguish two subtly different categories: discrete-time and piecewise constant.
+The discrete-time \lstinline!Real! variables must be assigned in a when-clause.
+The discrete-time variables can be used as argument to \lstinline!pre!.
+The discrete-time variables are a subset of all variables that are piecewise constant and only change at events.
+Tools may optimize code to only compute piecewise constant variables at events.
+
+\begin{lstlisting}[language=modelica]
+model ShowDiscrete
```

Comment:
I'm not if the _show_ in the model name really means anything.  Maybe `PiecewiseConstantReals` would capture more of the essence?

Reply:
Fine